python: adbc_driver_manager fails to build under conda on macOS

### What happened?

When trying to build the docs using the instructions in https://github.com/apache/arrow-adbc/blob/main/CONTRIBUTING.md#documentation, I get a build failure when I get to,

```sh
pip install -e .[test] # from python/adbc_driver_manager
```

I think the key error is,

> clang++ -fno-strict-overflow -Wsign-compare -Wunreachable-code -DNDEBUG -O2 -Wall -fPIC -O2 -isystem /opt/homebrew/Caskroom/miniforge/base/envs/adbc/include -arch arm64 -fPIC -O2 -isystem /opt/homebrew/Caskroom/miniforge/base/envs/adbc/include -arch arm64 -bundle -undefined dynamic_lookup -Wl,-rpath,/opt/homebrew/Caskroom/miniforge/base/envs/adbc/lib -L/opt/homebrew/Caskroom/miniforge/base/envs/adbc/lib -Wl,-rpath,/opt/homebrew/Caskroom/miniforge/base/envs/adbc/lib -L/opt/homebrew/Caskroom/miniforge/base/envs/adbc/lib build/temp.macosx-11.0-arm64-cpython-313/adbc_driver_manager/_backward.o -o build/lib.macosx-11.0-arm64-cpython-313/adbc_driver_manager/_backward.cpython-313-darwin.so
>      ld: -lto_library library filename must be 'libLTO.dylib'
>      clang++: error: linker command failed with exit code 1 (use -v to see invocation)
<details>
<summary>Full Output</summary>

```sh
adbc ❯ pip install .[test]                                                                                                                                                                               (adbc)
Processing /Users/bryce/src/apache/arrow-adbc/python/adbc_driver_manager
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: typing-extensions in /opt/homebrew/Caskroom/miniforge/base/envs/adbc/lib/python3.13/site-packages (from adbc-driver-manager==1.7.0.dev0) (4.14.0)
Collecting duckdb (from adbc-driver-manager==1.7.0.dev0)
  Using cached duckdb-1.3.1-cp313-cp313-macosx_12_0_arm64.whl.metadata (7.0 kB)
Collecting pandas (from adbc-driver-manager==1.7.0.dev0)
  Using cached pandas-2.3.0-cp313-cp313-macosx_11_0_arm64.whl.metadata (91 kB)
Requirement already satisfied: polars in /opt/homebrew/Caskroom/miniforge/base/envs/adbc/lib/python3.13/site-packages (from adbc-driver-manager==1.7.0.dev0) (1.31.0)
Collecting pyarrow>=14.0.1 (from adbc-driver-manager==1.7.0.dev0)
  Using cached pyarrow-20.0.0-cp313-cp313-macosx_12_0_arm64.whl.metadata (3.3 kB)
Requirement already satisfied: pytest in /opt/homebrew/Caskroom/miniforge/base/envs/adbc/lib/python3.13/site-packages (from adbc-driver-manager==1.7.0.dev0) (8.4.1)
Requirement already satisfied: numpy>=1.26.0 in /opt/homebrew/Caskroom/miniforge/base/envs/adbc/lib/python3.13/site-packages (from pandas->adbc-driver-manager==1.7.0.dev0) (2.3.1)
Requirement already satisfied: python-dateutil>=2.8.2 in /opt/homebrew/Caskroom/miniforge/base/envs/adbc/lib/python3.13/site-packages (from pandas->adbc-driver-manager==1.7.0.dev0) (2.9.0.post0)
Requirement already satisfied: pytz>=2020.1 in /opt/homebrew/Caskroom/miniforge/base/envs/adbc/lib/python3.13/site-packages (from pandas->adbc-driver-manager==1.7.0.dev0) (2025.2)
Collecting tzdata>=2022.7 (from pandas->adbc-driver-manager==1.7.0.dev0)
  Using cached tzdata-2025.2-py2.py3-none-any.whl.metadata (1.4 kB)
Requirement already satisfied: six>=1.5 in /opt/homebrew/Caskroom/miniforge/base/envs/adbc/lib/python3.13/site-packages (from python-dateutil>=2.8.2->pandas->adbc-driver-manager==1.7.0.dev0) (1.17.0)
Requirement already satisfied: iniconfig>=1 in /opt/homebrew/Caskroom/miniforge/base/envs/adbc/lib/python3.13/site-packages (from pytest->adbc-driver-manager==1.7.0.dev0) (2.0.0)
Requirement already satisfied: packaging>=20 in /opt/homebrew/Caskroom/miniforge/base/envs/adbc/lib/python3.13/site-packages (from pytest->adbc-driver-manager==1.7.0.dev0) (25.0)
Requirement already satisfied: pluggy<2,>=1.5 in /opt/homebrew/Caskroom/miniforge/base/envs/adbc/lib/python3.13/site-packages (from pytest->adbc-driver-manager==1.7.0.dev0) (1.6.0)
Requirement already satisfied: pygments>=2.7.2 in /opt/homebrew/Caskroom/miniforge/base/envs/adbc/lib/python3.13/site-packages (from pytest->adbc-driver-manager==1.7.0.dev0) (2.19.2)
Using cached pyarrow-20.0.0-cp313-cp313-macosx_12_0_arm64.whl (30.8 MB)
Using cached duckdb-1.3.1-cp313-cp313-macosx_12_0_arm64.whl (15.5 MB)
Using cached pandas-2.3.0-cp313-cp313-macosx_11_0_arm64.whl (10.7 MB)
Using cached tzdata-2025.2-py2.py3-none-any.whl (347 kB)
Building wheels for collected packages: adbc-driver-manager
  Building wheel for adbc-driver-manager (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for adbc-driver-manager (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [85 lines of output]
      /private/var/folders/gf/3btcn6956190fx_xld0y_b9r0000gn/T/pip-build-env-lmvyzaan/overlay/lib/python3.13/site-packages/setuptools/dist.py:332: InformationOnly: Normalizing '1.7.0dev' to '1.7.0.dev0'
        self.metadata.version = self._normalize_version(self.metadata.version)
      /private/var/folders/gf/3btcn6956190fx_xld0y_b9r0000gn/T/pip-build-env-lmvyzaan/overlay/lib/python3.13/site-packages/setuptools/config/_apply_pyprojecttoml.py:82: SetuptoolsDeprecationWarning: `project.license` as a TOML table is deprecated
      !!

              ********************************************************************************
              Please use a simple string containing a SPDX expression for `project.license`. You can also use `project.license-files`. (Both options available on setuptools>=77.0.0).

              By 2026-Feb-18, you need to update your project and remove deprecated calls
              or your builds will no longer be supported.

              See https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#license for details.
              ********************************************************************************

      !!
        corresp(dist, value, root_dir)
      /private/var/folders/gf/3btcn6956190fx_xld0y_b9r0000gn/T/pip-build-env-lmvyzaan/overlay/lib/python3.13/site-packages/setuptools/config/_apply_pyprojecttoml.py:55: SetuptoolsDeprecationWarning: 'tool.setuptools.license-files' is deprecated in favor of 'project.license-files' (available on setuptools>=77.0.0).
      !!

              ********************************************************************************

              By 2026-Feb-18, you need to update your project and remove deprecated calls
              or your builds will no longer be supported.

              See https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#license-files for details.
              ********************************************************************************

      !!
        _apply_tool_table(dist, config, filename)
      running bdist_wheel
      running build
      running build_py
      file adbc_driver_manager.py (for module adbc_driver_manager) not found
      creating build/lib.macosx-11.0-arm64-cpython-313/adbc_driver_manager
      copying adbc_driver_manager/_version.py -> build/lib.macosx-11.0-arm64-cpython-313/adbc_driver_manager
      copying adbc_driver_manager/dbapi.py -> build/lib.macosx-11.0-arm64-cpython-313/adbc_driver_manager
      copying adbc_driver_manager/__init__.py -> build/lib.macosx-11.0-arm64-cpython-313/adbc_driver_manager
      copying adbc_driver_manager/_static_version.py -> build/lib.macosx-11.0-arm64-cpython-313/adbc_driver_manager
      running egg_info
      writing adbc_driver_manager.egg-info/PKG-INFO
      writing dependency_links to adbc_driver_manager.egg-info/dependency_links.txt
      writing requirements to adbc_driver_manager.egg-info/requires.txt
      writing top-level names to adbc_driver_manager.egg-info/top_level.txt
      file adbc_driver_manager.py (for module adbc_driver_manager) not found
      reading manifest file 'adbc_driver_manager.egg-info/SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      adding license file 'LICENSE.txt'
      adding license file 'NOTICE.txt'
      writing manifest file 'adbc_driver_manager.egg-info/SOURCES.txt'
      copying adbc_driver_manager/_backward.pyx -> build/lib.macosx-11.0-arm64-cpython-313/adbc_driver_manager
      copying adbc_driver_manager/_blocking_impl.cc -> build/lib.macosx-11.0-arm64-cpython-313/adbc_driver_manager
      copying adbc_driver_manager/_blocking_impl.h -> build/lib.macosx-11.0-arm64-cpython-313/adbc_driver_manager
      copying adbc_driver_manager/_lib.pxd -> build/lib.macosx-11.0-arm64-cpython-313/adbc_driver_manager
      copying adbc_driver_manager/_lib.pyi -> build/lib.macosx-11.0-arm64-cpython-313/adbc_driver_manager
      copying adbc_driver_manager/_lib.pyx -> build/lib.macosx-11.0-arm64-cpython-313/adbc_driver_manager
      copying adbc_driver_manager/_reader.pyi -> build/lib.macosx-11.0-arm64-cpython-313/adbc_driver_manager
      copying adbc_driver_manager/_reader.pyx -> build/lib.macosx-11.0-arm64-cpython-313/adbc_driver_manager
      copying adbc_driver_manager/adbc_driver_manager.cc -> build/lib.macosx-11.0-arm64-cpython-313/adbc_driver_manager
      copying adbc_driver_manager/backward.hpp -> build/lib.macosx-11.0-arm64-cpython-313/adbc_driver_manager
      copying adbc_driver_manager/py.typed -> build/lib.macosx-11.0-arm64-cpython-313/adbc_driver_manager
      creating build/lib.macosx-11.0-arm64-cpython-313/adbc_driver_manager/arrow-adbc
      copying adbc_driver_manager/arrow-adbc/adbc.h -> build/lib.macosx-11.0-arm64-cpython-313/adbc_driver_manager/arrow-adbc
      copying adbc_driver_manager/arrow-adbc/adbc_driver_manager.h -> build/lib.macosx-11.0-arm64-cpython-313/adbc_driver_manager/arrow-adbc
      file adbc_driver_manager.py (for module adbc_driver_manager) not found
      running build_ext
      Compiling adbc_driver_manager/_backward.pyx because it changed.
      [1/1] Cythonizing adbc_driver_manager/_backward.pyx
      building 'adbc_driver_manager._backward' extension
      creating build/temp.macosx-11.0-arm64-cpython-313/adbc_driver_manager
      clang++ -fno-strict-overflow -Wsign-compare -Wunreachable-code -DNDEBUG -O2 -Wall -fPIC -O2 -isystem /opt/homebrew/Caskroom/miniforge/base/envs/adbc/include -arch arm64 -fPIC -O2 -isystem /opt/homebrew/Caskroom/miniforge/base/envs/adbc/include -arch arm64 -I/Users/bryce/src/apache/arrow-adbc/python/adbc_driver_manager/adbc_driver_manager -I/opt/homebrew/Caskroom/miniforge/base/envs/adbc/include/python3.13 -c adbc_driver_manager/_backward.cpp -o build/temp.macosx-11.0-arm64-cpython-313/adbc_driver_manager/_backward.o -std=c++17
      adbc_driver_manager/_backward.cpp:1540:3: warning: struct 'SignalHandling' was previously declared as a class; this is valid, but may result in linker errors under the Microsoft C++ ABI [-Wmismatched-tags]
       1540 |   struct backward::SignalHandling _c_signal_handler;
            |   ^
      adbc_driver_manager/backward.hpp:4174:7: note: previous use is here
       4174 | class SignalHandling {
            |       ^
      adbc_driver_manager/_backward.cpp:1540:3: note: did you mean class here?
       1540 |   struct backward::SignalHandling _c_signal_handler;
            |   ^~~~~~
            |   class
      1 warning generated.
      clang++ -fno-strict-overflow -Wsign-compare -Wunreachable-code -DNDEBUG -O2 -Wall -fPIC -O2 -isystem /opt/homebrew/Caskroom/miniforge/base/envs/adbc/include -arch arm64 -fPIC -O2 -isystem /opt/homebrew/Caskroom/miniforge/base/envs/adbc/include -arch arm64 -bundle -undefined dynamic_lookup -Wl,-rpath,/opt/homebrew/Caskroom/miniforge/base/envs/adbc/lib -L/opt/homebrew/Caskroom/miniforge/base/envs/adbc/lib -Wl,-rpath,/opt/homebrew/Caskroom/miniforge/base/envs/adbc/lib -L/opt/homebrew/Caskroom/miniforge/base/envs/adbc/lib build/temp.macosx-11.0-arm64-cpython-313/adbc_driver_manager/_backward.o -o build/lib.macosx-11.0-arm64-cpython-313/adbc_driver_manager/_backward.cpython-313-darwin.so
      ld: -lto_library library filename must be 'libLTO.dylib'
      clang++: error: linker command failed with exit code 1 (use -v to see invocation)
      error: command '/opt/homebrew/Caskroom/miniforge/base/envs/adbc/bin/clang++' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for adbc-driver-manager
Failed to build adbc-driver-manager
ERROR: Failed to build installable wheels for some pyproject.toml based projects (adbc-driver-manager)
```
</details>

I can build successfully:

- Under conda on my amd64 Linux machine
- Under a venv on macOS (i.e., just not conda)

### Stack Trace

_No response_

### How can we reproduce the bug?

_No response_

### Environment/Setup

_No response_

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -220,32 +220,57 @@ $ npm install -g @mermaid-js/mermaid-cli
 To build the HTML documentation:
 
 ```shell
-$ pushd c/apidoc
-$ doxygen
-$ popd
-
-# Optionally: to also build the Python documentation
-$ pushd python/adbc_driver_manager
-$ pip install -e .[test]
-$ popd
-
 $ pushd docs
 $ make html
 ```
 
-The output can be found in `build/`.
+The output can be found in `build/`.  This does not generate API references
+and results in some warnings, but it is not a problem if you're not working
+with the API documentation.
 
 Some documentations are maintained as [Mermaid][mermaid] diagrams, which must
 be rendered and checked in.  This can be done as follows:
 
 ```shell
-pushd docs
-make -f mermaid.makefile -j all
+$ pushd docs
+$ make -f mermaid.makefile -j all
 # Check in the updated files
 ```
 
 [mermaid]: https://mermaid.js.org/
 [sphinx]: https://www.sphinx-doc.org/en/master/
+
+#### Building more complete documentation
+
+You can remove the warnings of `make html` and generate the Python API
+reference as follows:
+
+```shell
+$ mamba create -n adbc \
+  --file ci/conda_env_docs.txt \
+  --file ci/conda_env_cpp.txt \
+  --file ci/conda_env_python.txt \
+  --file ci/conda_env_java.txt
+$ mamba activate adbc
+$ env ADBC_USE_ASAN=0 ADBC_USE_UBSAN=0 ./ci/scripts/python_build.sh $(pwd) $(pwd)/build
+$ pushd docs
+$ make html
+```
+
+For a more complete build, you can use the following script:
+
+```shell
+$ ./ci/scripts/docs_build.sh "$(pwd)"
+```
+
+This generates all available API references, and also runs doctests.
+
+To generate the R API reference, you need to run the following additionally:
+
+```shell
+$ mamba install --file ci/conda_env_r.txt
+$ ./ci/scripts/r_build.sh $(pwd)
+```
 
 ### GLib
 

